ros2_graph/graph_generator.py

```diff
@@ -158,29 +168,46 @@ def nodes_from_data(self, nodes_data: Tuple[Tuple[str]]) -> Tuple[NodeElement]:
         @param nodes_data: nodes info in (name, namespace) format
         @return a tuple of NodeElement objects
         """
-        return tuple((self.new_node(*node) for node in nodes_data))
+        nodes = (self.new_node(*node) for node in nodes_data)
+        return tuple((node for node in nodes if node is not None))
 
     def filter_topics(self, topic: Tuple[str, str]) -> bool:
         """! To filter undesired topics
         @param topic tuple(name, type)
         @return False if the topic is not desired
         """
-        exclude = [
-            "/parameter_events",
-            "/rosout",
-            "/tf",
-            "/cascade_lifecycle_activations",
-            "/cascade_lifecycle_states",
-            "",
-        ]
-
         blacklist = ("/transition_event", "/_action")
 
-        if topic[0] in exclude:
+        if topic[0] in self.to_ignore["topics"]:
             return False
 
         return is_not_in_blacklist(topic, blacklist)
 
+    def filter_services(self, service: Tuple[str, str]) -> bool:
+        """! To filter undesired topics
+        @param service tuple(name, type)
+        @return False if the service is not desired
+        """
+        blacklist = (
+            "/change_state",
+            "/describe_parameters",
+            "/get_available_states",
+            "/get_available_transitions",
+            "/get_parameter_types",
+            "/get_parameters",
+            "/get_state",
+            "/get_transition_graph",
+            "/list_parameters",
+            "/set_parameters",
+            "/set_parameters_atomically",
+            "/_action",
+        )
```

Comment:
why not moving this where you define `self.to_ignore`?

Reply:
Done